substra/sdk/client.py

```diff
@@ -76,39 +79,48 @@ def __init__(
         token: Optional[str] = None,
         retry_timeout: int = DEFAULT_RETRY_TIMEOUT,
         insecure: bool = False,
-        debug: bool = False,
+        backend_type: schemas.BackendType = schemas.BackendType.REMOTE,
     ):
         self._retry_timeout = retry_timeout
         self._token = token
 
         self._insecure = insecure
         self._url = url
 
-        self._backend = self._get_backend(debug)
+        self._backend = self._get_backend(backend_type)
 
-    def _get_backend(self, debug: bool):
+    def _get_backend(self, backend_type: schemas.BackendType):
         # Three possibilities:
-        # - debug is False: get a remote backend
-        # - debug is True and no url is defined: fully local backend
-        # - debug is True and url is defined: local backend that connects to
-        #                           a remote backend (read-only)
-        backend = None
-        if (debug and self._url) or not debug:
-            backend = backends.get(
-                "remote",
+        # - deployed: get a deployed backend
+        # - subprocess/docker and no url is defined: fully local backend
+        # - subprocess/docker and url is defined: local backend that connects to
+        #                           a deployed backend (read-only)
+        if backend_type == schemas.BackendType.REMOTE:
+            return backends.get(
+                schemas.BackendType.REMOTE,
                 url=self._url,
                 insecure=self._insecure,
                 token=self._token,
                 retry_timeout=self._retry_timeout,
             )
-        if debug:
-            # Hybrid mode: the local backend also connects to
-            # a remote backend in read-only mode.
+        if backend_type in [schemas.BackendType.LOCAL_DOCKER, schemas.BackendType.LOCAL_SUBPROCESS]:
```

Comment:
```suggestion
        elif backend_type in [schemas.BackendType.LOCAL_DOCKER, schemas.BackendType.LOCAL_SUBPROCESS]:
```

Reply:
we do not need a `elif` since there is a `return` statement in the `if` above